### PR TITLE
#57 Improve the user interface of compileModelicaOMC

### PR DESCRIPTION
**Pull request by rosiereflo**

### CODE REVIEW

**Comment by rosiereflo** [comment]
An update of the documentation will be required

File: dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp [hunk 1]
```diff
@@ -150,11 +158,49 @@ int main(int argc, char ** argv) {
...
 }
 
+
+void
+prepareWorkspace(const string& modelName, const string& inputDir, const string& outputDir, bool& rmModels) {
+  if (!is_directory(outputDir))
+    create_directory(outputDir);
+  string outputDir1 = prettyPath(outputDir);
+  if (!is_directory(inputDir))
+    throw DYNError(DYN::Error::MODELER, MissingModelicaFile, absolute(modelName + ".mo", inputDir));
```
Comment (adriguir) [inline dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp]: I would have changed the error message to create a more specific one (even if it is correct, there is no Modelica model input but that is a bit worst -> there are no input data at the specified place)

File: dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp [hunk 2]
```diff
@@ -352,10 +398,12 @@ compileLib(const string& modelName, const string& libName, const string& outputD
 }
 
 void
-removeTemporaryFiles(const string& modelName, const string& outputDir) {
+removeTemporaryFiles(const string& modelName, const string& outputDir, bool rmModels) {
   string outputDir1 = prettyPath(outputDir);
   string scriptsDir1 = getEnvVar("DYNAWO_SCRIPTS_DIR");
   string commandRemove = scriptsDir1 + "/cleanCompilerModelicaOMC --model=" + modelName + " --directory=" + outputDir1;
+  if (!rmModels)
+    commandRemove += " --do-not-remove-model-files";
```
Comment (adriguir) [inline dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp]: I find it a bif confusing to change the logic there. When you look at the whole code, the first part is working in a remove files way, the second one in a not remove files way (always confusing when coming back to such a development later onà)

File: dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp]: Solved

File: dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp (hunk 2 above)
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/DYNCompilerModelicaOMC.cpp]: I proposed something else in the latest commit, please check if this new solution is OK with you

**Pull request merged**